From 640a2510791dd4f6ad295a3f529b516092fd4e31 Mon Sep 17 00:00:00 2001
From: Phil Staub <Phil.Staub@windriver.com>
Date: Thu, 24 Sep 2009 17:35:03 -0700
Subject: [PATCH] MIPS: Enable building Cavium I2C support

Hooks in Makefile and Kconfig to allow selection/building of the I2C
support.

The current state of these files reflects an ongoing development
process, and as such implies an extensive modification history that is
not cited in detail here. It is worth noting, however, that this
collection represents the work of the following people who were
involved in that history:

David Daney <ddaney@caviumnetworks.com>
Tomaso Paoletti <tpaoletti@caviumnetworks.com>
Vlad Malov <vlad.malov@caviumnetworks.com>
Paul Gortmaker <Paul.Gortmaker@windriver.com>
Phil Staub <Phil.Staub@windriver.com>

Signed-off-by: Phil Staub <Phil.Staub@windriver.com>
---
 drivers/i2c/busses/Kconfig  |   17 +++++++++++++++++
 drivers/i2c/busses/Makefile |    3 +++
 2 files changed, 20 insertions(+), 0 deletions(-)

diff --git a/drivers/i2c/busses/Kconfig b/drivers/i2c/busses/Kconfig
index eb94378..dcae2ee 100644
--- a/drivers/i2c/busses/Kconfig
+++ b/drivers/i2c/busses/Kconfig
@@ -716,4 +716,21 @@ config SCx200_ACB
 	  This support is also available as a module.  If so, the module
 	  will be called scx200_acb.
 
+config I2C_OCTEON_TWSI
+	tristate "Octeon TWSI-I2C"
+	depends on I2C
+	help
+	  Octeon TWSI-I2C
+
+	  This driver can also be built as a module.  If so, the module
+	  will be called i2c-octeon-twsi.
+
+config I2C_OCTEON_TWSI_RAW
+	tristate "Octeon TWSI-I2C-RAW"
+	depends on I2C_OCTEON_TWSI
+	help
+	  Octeon TWSI-I2C low-lewel interface
+
+	  This is additional low-lewel interface to Octeon twsi controller. Do not build it
+	  unless i2c transactions longer than 8 data bytes will be required. 
 endmenu
diff --git a/drivers/i2c/busses/Makefile b/drivers/i2c/busses/Makefile
index d4dc22d..092631e 100644
--- a/drivers/i2c/busses/Makefile
+++ b/drivers/i2c/busses/Makefile
@@ -46,6 +46,8 @@ obj-$(CONFIG_I2C_SH7760)	+= i2c-sh7760.o
 obj-$(CONFIG_I2C_SH_MOBILE)	+= i2c-sh_mobile.o
 obj-$(CONFIG_I2C_SIMTEC)	+= i2c-simtec.o
 obj-$(CONFIG_I2C_VERSATILE)	+= i2c-versatile.o
+obj-$(CONFIG_I2C_OCTEON_TWSI)	+= i2c-oct.o
+i2c-oct-objs := i2c-octeon.o ../../../arch/mips/cavium-octeon/i2c-octeon_twsi.o ../../../arch/mips/cavium-octeon/i2c-octeon_twsi_raw.o
 
 # External I2C/SMBus adapter drivers
 obj-$(CONFIG_I2C_PARPORT)	+= i2c-parport.o
@@ -71,3 +73,4 @@ obj-$(CONFIG_I2C_OMAP)          += i2c-omap.o
 ifeq ($(CONFIG_I2C_DEBUG_BUS),y)
 EXTRA_CFLAGS += -DDEBUG
 endif
+
-- 
1.5.5.1

